python/fbprophet/forecaster.py

```diff
@@ -1097,6 +1097,31 @@ def stan_init():
                 'sigma_obs': 1,
             }
 
+        cmdstanpy_data = {
```

Comment:
Can we either subclass `Forecaster` and add this as an alternative implementation, or delegate this new code path to a different method.

# Option 1

```
class CmdStanPyForecaster(Forecaster):
    def fit(): ...
        # new fit implementation
```
# Option 2

```
class Forecaster: # existing class 
    def __init__(self, ..., stan_backend='pystan'):
        self.stan_backend = stan_backend
    def fit(...):
        if self.stan_backend == 'cmdstanpy':
            self.fit_cmdstanpy(...)
        else:
            self.fit_pystan(...)
```

# Option 3

``` 
class Forecaster: # existing class
    def fit(..., stan_backend='pystan'):
        # similar to option 2
```